internal/generator/vector/output/kafka/sasl.go

```diff
@@ -1,34 +1,29 @@
 package kafka
 
-import (
-	"fmt"
-
-	"github.com/openshift/cluster-logging-operator/internal/generator"
+const (
+	SASLMechanismPlain = "PLAIN"
+	SASLMechamisnSSL   = "SCRAM-SHA-256"
 )
 
-type Sasl bool
-
-func (s Sasl) Name() string {
-	return "kafkaSaslTemplate"
+type Sasl struct {
+	Desc        string
+	ComponentID string
+	Username    string
+	Password    string
+	Mechanism   string
 }
 
-func (s Sasl) Template() string {
-	return fmt.Sprintf(`{{define "kafkaSaslTemplate" -}}
-enabled = %t
-{{end}}
-`, s)
-}
-
-type SaslConf generator.ConfLiteral
-
-func (t SaslConf) Name() string {
-	return "kafkaSasl"
+func (t Sasl) Name() string {
+	return "vectorKafkaSasl"
 }
 
-func (t SaslConf) Template() string {
-	return `
-{{define "kafkaSasl" -}}
+func (t Sasl) Template() string {
+	return `{{define "vectorKafkaSasl"}}
 # {{.Desc}}
 [sinks.{{.ComponentID}}.sasl]
-{{- end}}`
+enabled = true
+username = "{{.Username}}"
```

Comment:
What happens if username/password are missing/empty? Is `username = ""` the same as no username?

Reply:
this configuration is mainly for `SASL/PLAIN` , which needs username/password
We could add a verification step to check username/password is always configured in secret, if `SASL/PLAIN` is configured